src/placeos-models/alert.cr

```diff
@@ -0,0 +1,68 @@
+require "json"
+require "./base/model"
+require "./trigger/conditions"
+
+module PlaceOS::Model
+  class Alert < ModelBase
+    include PlaceOS::Model::Timestamps
+
+    table :alert
+
+    enum Severity
+      LOW
+      MEDIUM
+      HIGH
+      CRITICAL
+    end
+
+    enum AlertType
+      THRESHOLD
+      STATUS
+      CUSTOM
+    end
+
+    attribute name : String, es_subfield: "keyword"
+    attribute description : String = ""
+    attribute enabled : Bool = true
+
+    # Reuse the same conditions structure as Trigger
+    attribute conditions : PlaceOS::Model::Trigger::Conditions = -> { PlaceOS::Model::Trigger::Conditions.new }, es_ignore: true
+
+    attribute severity : Severity = Severity::MEDIUM, converter: PlaceOS::Model::PGEnumConverter(PlaceOS::Model::Alert::Severity)
+    attribute alert_type : AlertType = AlertType::THRESHOLD, converter: PlaceOS::Model::PGEnumConverter(PlaceOS::Model::Alert::AlertType)
+
+    # In milliseconds - how often to check the condition
+    attribute check_interval : Int32 = 60000 # 1 minute default
```

Comment:
I don't think we need check_interval as MQTT pushes data as it changes so the frontent will be notified of any changes as soon as they occur.

Hence the conditions can be checked each time there is an update.

What might be useful is a delay on notification, like say don't show low priority issues unless they have been active for 1min without resolving automatically, 30 seconds for medium, 15 seconds for high and 2 seconds for critical (as defaults)
The user could configure alternative times - but the defaults might be simpler to be defined on the frontend and we just have the model default to 15 seconds if the frontend doesn't specify